ocaml/xapi/xapi_vbd_helpers.ml

```diff
@@ -262,15 +262,6 @@ let update_allowed_operations ~__context ~self : unit =
   let keys = Hashtbl.fold (fun k v acc -> if v = None then k :: acc else acc) valid [] in
   Db.VBD.set_allowed_operations ~__context ~self ~value:keys
 
-(** Someone is cancelling a task so remove it from the current_operations *)
-let cancel_task ~__context ~self ~task_id =
-  let all = List.map fst (Db.VBD.get_current_operations ~__context ~self) in
-  if List.mem task_id all then
-    begin
-      Db.VBD.remove_from_current_operations ~__context ~self ~key:task_id;
-      update_allowed_operations ~__context ~self
```

Comment:
Again, a suspicious lack of `update_allowed_operations` in the `cancel_tasks` function below.

Reply:
It also comes in by the Initial import from hg on 2009 and no changes after that.

Reply:
See above.